### PR TITLE
[pipelineX](refactor) add repeat node in pipelineX

### PR DESCRIPTION
**Pull request by Mryange**
## Proposed changes

Issue Number: close #xxx

## Further comments

If this is a relatively large or complex change, kick off the discussion at [dev@doris.apache.org](mailto:dev@doris.apache.org) by explaining why you chose the solution you did and what alternatives you considered, etc...

### CODE REVIEW

**Comment by Mryange** [comment]
run buildall

File: be/src/pipeline/exec/repeat_operator.h [hunk 1]
```diff
@@ -45,5 +45,56 @@ class RepeatOperator final : public StatefulOperator<RepeatOperatorBuilder> {
...
+    std::set<SlotId> _all_slot_ids;
+    // An integer bitmap list, it indicates the bit position of the exprs not null.
+    std::vector<int64_t> _repeat_id_list;
+    std::vector<std::vector<int64_t>> _grouping_list;
+    TupleId _output_tuple_id;
+    const TupleDescriptor* _output_tuple_desc;
+
+    std::vector<SlotDescriptor*> _output_slots;
+
+    vectorized::VExprContextSPtrs _expr_ctxs;
```
Comment (Gabriel39) [inline be/src/pipeline/exec/repeat_operator.h:96]: Also need this in local state

File: be/src/pipeline/exec/repeat_operator.h [hunk 2]
```diff
@@ -45,5 +45,56 @@ class RepeatOperator final : public StatefulOperator<RepeatOperatorBuilder> {
...
+
+private:
+    friend class RepeatOperatorX;
+    std::unique_ptr<vectorized::Block> _child_block;
+    SourceState _child_source_state;
+    bool _child_eos;
+    int _repeat_id_idx;
+    std::unique_ptr<vectorized::Block> _intermediate_block {};
+};
+class RepeatOperatorX final : public OperatorXBase {
```
Comment (Gabriel39) [inline be/src/pipeline/exec/repeat_operator.h:65]:
```suggestion
class RepeatOperatorX final : public OperatorX<RepeatLocalState> {
```

**Comment by hello-stephen** [comment]
(From new machine)TeamCity pipeline, clickbench performance test result:
 the sum of best hot time: 50.24 seconds
 stream load tsv:          538 seconds loaded 74807831229 Bytes, about 132 MB/s
 stream load json:         20 seconds loaded 2358488459 Bytes, about 112 MB/s
 stream load orc:          65 seconds loaded 1101869774 Bytes, about 16 MB/s
 stream load parquet:          31 seconds loaded 861443392 Bytes, about 26 MB/s
 insert into select:          29.3 seconds inserted 10000000 Rows, about 341K ops/s
 storage size: 17162085143 Bytes

**Comment by Mryange** [comment]
run buildall

**Comment by hello-stephen** [comment]
(From new machine)TeamCity pipeline, clickbench performance test result:
 the sum of best hot time: 45.95 seconds
 stream load tsv:          535 seconds loaded 74807831229 Bytes, about 133 MB/s
 stream load json:         20 seconds loaded 2358488459 Bytes, about 112 MB/s
 stream load orc:          64 seconds loaded 1101869774 Bytes, about 16 MB/s
 stream load parquet:          30 seconds loaded 861443392 Bytes, about 27 MB/s
 insert into select:          28.8 seconds inserted 10000000 Rows, about 347K ops/s
 storage size: 17162120907 Bytes

File: be/src/pipeline/exec/repeat_operator.cpp [hunk 3]
```diff
@@ -42,4 +43,239 @@ Status RepeatOperator::close(doris::RuntimeState* state) {
...
+    RETURN_IF_ERROR(Base::init(state, info));
+    auto& p = _parent->cast<Parent>();
+    _expr_ctxs.resize(p._expr_ctxs.size());
+    for (size_t i = 0; i < _expr_ctxs.size(); i++) {
+        RETURN_IF_ERROR(p._expr_ctxs[i]->clone(state, _expr_ctxs[i]));
+    }
+    return Status::OK();
+}
+
+Status RepeatLocalState::close(RuntimeState* state) {
```
Comment (Gabriel39) [inline be/src/pipeline/exec/repeat_operator.cpp:63]: Do not need to override this function

File: be/src/pipeline/exec/repeat_operator.cpp [hunk 4]
```diff
@@ -42,4 +43,239 @@ Status RepeatOperator::close(doris::RuntimeState* state) {
...
+
+    return Status::OK();
+}
+
+Status RepeatOperatorX::open(RuntimeState* state) {
+    VLOG_CRITICAL << "VRepeatNode::open";
+    SCOPED_TIMER(_runtime_profile->total_time_counter());
+    RETURN_IF_ERROR(OperatorXBase::open(state));
+    RETURN_IF_ERROR(vectorized::VExpr::open(_expr_ctxs, state));
+    RETURN_IF_ERROR(_child_x->open(state));
```
Comment (Gabriel39) [inline be/src/pipeline/exec/repeat_operator.cpp:98]: This is already done in `OperatorXBase::open`

**Comment by Mryange** [comment]
run buildall

**Comment by hello-stephen** [comment]
(From new machine)TeamCity pipeline, clickbench performance test result:
 the sum of best hot time: 46.72 seconds
 stream load tsv:          528 seconds loaded 74807831229 Bytes, about 135 MB/s
 stream load json:         20 seconds loaded 2358488459 Bytes, about 112 MB/s
 stream load orc:          64 seconds loaded 1101869774 Bytes, about 16 MB/s
 stream load parquet:          31 seconds loaded 861443392 Bytes, about 26 MB/s
 insert into select:          29.0 seconds inserted 10000000 Rows, about 344K ops/s
 storage size: 17162119801 Bytes

**Comment by Mryange** [comment]
run buildall

File: be/src/pipeline/exec/repeat_operator.cpp [hunk 5]
```diff
@@ -37,9 +38,238 @@ Status RepeatOperator::prepare(doris::RuntimeState* state) {
...
+bool RepeatOperatorX::need_more_input_data(RuntimeState* state) const {
+    auto& local_state = state->get_local_state(id())->cast<RepeatLocalState>();
+    return !local_state._child_block->rows() && !local_state._child_eos;
+}
+Status RepeatOperatorX::get_block(RuntimeState* state, vectorized::Block* block,
+                                  SourceState& source_state) {
+    auto& local_state = state->get_local_state(id())->cast<RepeatLocalState>();
+    if (need_more_input_data(state)) {
+        local_state._child_block->clear_column_data();
+        RETURN_IF_ERROR(_child_x->get_block(state, local_state._child_block.get(),
```
Comment (Gabriel39) [inline be/src/pipeline/exec/repeat_operator.cpp:114]:
```suggestion
        RETURN_IF_ERROR(_child_x->get_next_after_projects(state, local_state._child_block.get(),
```

**Comment by Mryange** [comment]
run buildall

**Review by Gabriel39 — APPROVED** [review]
LGTM

**Comment by Mryange** [comment]
run clickbench

**Comment by hello-stephen** [comment]
(From new machine)TeamCity pipeline, clickbench performance test result:
 the sum of best hot time: 45.52 seconds
 stream load tsv:          539 seconds loaded 74807831229 Bytes, about 132 MB/s
 stream load json:         20 seconds loaded 2358488459 Bytes, about 112 MB/s
 stream load orc:          64 seconds loaded 1101869774 Bytes, about 16 MB/s
 stream load parquet:          31 seconds loaded 861443392 Bytes, about 26 MB/s
 insert into select:          29.0 seconds inserted 10000000 Rows, about 344K ops/s
 storage size: 17162053593 Bytes

**Review by zclllyybb — APPROVED** [review]

**Pull request merged**